hooks/useClaimableTree.ts

```diff
@@ -0,0 +1,60 @@
+import { StandardMerkleTree } from '@openzeppelin/merkle-tree';
+import { useMemo } from 'react';
+import useSWR from 'swr';
+import { zeroAddress } from 'viem';
+
+export type ClaimableItemLeaf = [bigint, `0x${string}`, bigint];
+
+type FetcherInput = [`0x${string}`, bigint];
+
+const fetcher = async ([
+  _gameAddress,
+  _itemId,
+]: FetcherInput): Promise<StandardMerkleTree<ClaimableItemLeaf> | null> => {
+  try {
+    const uri = `/api/getTree?gameAddress=${_gameAddress}&itemId=${_itemId.toString()}`;
+
+    const data = await fetch(uri);
+    const { tree } = await data.json();
+
+    if (!tree) {
+      return null;
+    }
+
+    const merkleTree = StandardMerkleTree.load(JSON.parse(tree));
+
+    return merkleTree as StandardMerkleTree<ClaimableItemLeaf>;
+  } catch (e) {
+    return null;
+  }
+};
+
+export const useClaimableTree = (
+  gameAddress: `0x${string}`,
+  itemId: bigint,
+): {
+  tree: StandardMerkleTree<ClaimableItemLeaf> | null;
+  reload: () => void;
+  loading: boolean;
+  error: Error | null;
+} => {
+  const input: FetcherInput = useMemo(
+    () => [gameAddress, itemId],
+    [gameAddress, itemId],
+  );
+
+  const { data, error, mutate, isLoading, isValidating } = useSWR<
+    StandardMerkleTree<ClaimableItemLeaf> | null,
+    Error,
+    FetcherInput
+  >(input, fetcher, {
+    isPaused: () => !gameAddress || !itemId || gameAddress === zeroAddress,
+  });
+
+  return {
+    loading: isLoading || isValidating || data === undefined,
```

Comment:
For some reason `loading` is hanging for a long time when the `ClaimItemModal` first pops up. Seems like it has to do with `data` not getting fetched immediately

Reply:
Yeah. I have seen this. I dont know why it hangs so long. Do you have any idea ?

Reply:
Going to try to debug it today